ome_zarr/reader.py

```diff
@@ -324,6 +337,30 @@ def array(self, resolution: str, version: str) -> da.core.Array:
         # data.shape is (t, c, z, y, x) by convention
         return self.zarr.load(resolution)
 
+    def validate(self, warnings: bool = False) -> None:
+        multiscales = self.lookup("multiscales", [])
+        version = multiscales[0].get("version", CurrentFormat().version)
+        LOGGER.info("Validating Multiscales spec at: %s" % self.zarr)
+        LOGGER.info("Using Multiscales schema version: %s" % version)
+        image_schema = get_schema(version)
+
+        # Always do a validation with the MUST rules
+        # Will throw ValidationException if it fails
+        json_data = self.zarr.root_attrs
+        jsonschema_validate(instance=json_data, schema=image_schema)
+
+        # If we're also checking for SHOULD rules,
+        # we want to iterate all errors and show as "Warnings"
+        if warnings:
+            strict_schema = get_schema(version, strict=True)
+            cls = validator_for(strict_schema)
+            cls.check_schema(strict_schema)
+            # Use our local resolver subclass to resolve local documents
+            localResolver = LocalRefResolver.from_schema(strict_schema)
+            validator = cls(strict_schema, resolver=localResolver)
+            for error in validator.iter_errors(json_data):
+                LOGGER.warn(error.message)
```

Comment:
sorry, not familiar with the code/design pattern here, can't see if there is some [MVC like pattern](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) where the warnings&errors (the Model) are collected/returned by some `validate` (the Controller), and then displayed (by the Viewer).
For me, as a 3rd party user intending to use this library to validate it, would be useful to have clear separation of the Model/Controller from the Viewer which would be "mine".

Here https://github.com/dandi/dandi-cli/issues/943#issue-1175862331 we are to "converge" on some data structure which we would use to cover validation reports across multiple validators possibly used for any specific file/dataset at hands.  So it would be important for us to be able to collect errors/warnings/hints uniformly first and then report them via our own "viewer".

Reply:
Thanks for your feedback. This work is at quite an early stage, so we haven't considered the usage of this tool outside of a cli validation. In fact, with changes to the schemas at https://github.com/ome/ngff/tree/main/0.4/schemas (include regular and "strict" schemas), there should be less logic required to validate.
You may find that you can choose the JSON schema validation tool of your choice, and don't need anything from `ome-zarr-py`.
In any case, this PR needs to be reviewed... 
cc @sbesson